src/wp-includes/blocks.php

```diff
@@ -178,10 +178,80 @@ function register_block_style_handle( $metadata, $field_name ) {
 	return $result ? $style_handle : false;
 }
 
+/**
+ * Gets i18n schema for block's metadata read from `block.json` file.
+ *
+ * @since 5.9.0
+ *
+ * @return array The schema for block's metadata.
+ */
+function get_i18n_block_metadata_schema() {
+	return array(
```

Comment:
@swissspidy, do you think it's fine to use PHP for the i18n schema used with `block.json` or should we copy the file from Gutenberg through WP packages (https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/src/api/i18n-block.json)?

Reply:
I'm not sure I follow... Does this now work differently from `theme-i18n.json`? Why not using `block-i18n.json` as planned?
	
The idea was for WP-CLI to have access to the schema via some publicly available URL, so that there's a way to easily get the latest version. For example `https://raw.githubusercontent.com/WordPress/wordpress-develop/master/src/wp-includes/theme-i18n.json` would already work.

Ideally this works the same for `block.json` as for `theme.json`. See `WP_Theme::translate()` for how it's done there. Surely there's a way to make this code reusable?!

Reply:
It works the same, but it uses a bit different logic. `WP_Theme::translate()` uses an intermediate step that produces a list of individual entries first which isn't necessary for the approach I reused from JavaScript. 

Anyway, I think I can make it work with JSON copied from Gutenberg. The only challenge is that I probably will need to parse JSON in a way that converts objects to `stdClass`. I will also check if we can move `WP_Theme::translate()` out of `WP_Theme` instead.

Reply:
I extracted `wp_json_file_decode` function to share the logic for reading and decoding JSON files. It made is simpler to use the `block-i18n.json` file copied from Gutenberg when translating block metadata.